substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -720,22 +836,21 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             let address = ensure_signed(origin)?;
 
-            let farm_name = FarmNameOf::<T>::try_from(name.clone().to_vec())
-                .map_err(DispatchErrorWithPostInfo::from)?;
+            let mut id = FarmID::<T>::get();
+            id = id + 1;
```

Comment:
why did you move this up?

Reply:
to follow the parameters order of the following Farm struct to be filled:
```rust

            let new_farm = Farm {
                version: TFGRID_FARM_VERSION,
                id,
                twin_id,
                name: farm_name,
                pricing_policy_id: 1,
                certification: FarmCertification::NotCertified,
                public_ips: public_ips_list,
                dedicated_farm: false,
                farming_policy_limits: None,
            };
```
There is no functional reason, I can go back